Order the types and method/attribute definitions generated by Orthoses::Trace

hibariya opened the pull request:
Since the Orthoses::Trace build type definitions from runtime information, they could be ordered differently depending on how they were called. I want to make it a bit more stable to make it less noisy and debugging easier.

ksss commented:
Thank you for using this library.
Please explain your issue in detail. 
Are you concerned about the order of the Union types?

ksss commented:
I see, I understand your request.
This change has the advantage of stabilizing the diff, but it also has the drawback of making it unclear in which order the calls were made. This seems to be a trade-off.
Orthoses values customizability. How about the idea of allowing the option to sort or not to be selected externally, with sorting being the default behavior?

Additionally, the method sort order can be resolved by using the `Orthoses::Sort` middleware.

hibariya commented:
@ksss Thanks for the feedback! I pushed another commit [here](https://github.com/ksss/orthoses/pull/79/commits/3eeea55a0410fc1d4fb9b3e7b7f94815aed358e2). Could you take a look when you have time? Thanks :pray:

> Orthoses values customizability. How about the idea of allowing the option to sort or not to be selected externally, with sorting being the default behavior?

Sounds good! I added the option keyword `sort_union_types: true`.

> Additionally, the method sort order can be resolved by using the Orthoses::Sort middleware.

Good point 👍  I removed that part from this PR.

ksss reviewed (approved):
GREAT! Thanks!
